pkg/stanza/fileconsumer/matcher/internal/filter/sort.go

```diff
@@ -130,3 +131,50 @@ func SortTemporal(regexKey string, ascending bool, layout string, location strin
 		},
 	)
 }
+
+type mtimeSortOption struct{}
+
+func newMtimeSortOption() Option {
+	return mtimeSortOption{}
+}
+
+type mtimeItem struct {
+	mtime time.Time
+	path  string
+	item  *item
+}
+
+func (m mtimeSortOption) apply(items []*item) ([]*item, error) {
+	mtimeItems := make([]mtimeItem, 0, len(items))
+	var errs error
+	for _, item := range items {
+		path := item.value
+		fi, err := os.Stat(path)
+		if err != nil {
+			errs = multierr.Append(errs, err)
+			continue
+		}
+
+		mtimeItems = append(mtimeItems, mtimeItem{
+			mtime: fi.ModTime(),
+			path:  path,
+			item:  item,
+		})
+	}
+
+	sort.SliceStable(mtimeItems, func(i, j int) bool {
+		// This checks if item i > j, in order to reverse the sort (most recently modified file is first in the list)
+		return mtimeItems[i].mtime.After(mtimeItems[j].mtime)
+	})
+
+	filteredValues := make([]*item, 0, len(items))
+	for _, mtimeItem := range mtimeItems {
+		filteredValues = append(filteredValues, mtimeItem.item)
+	}
+
+	return filteredValues, errs
+}
+
+func SortMtime() Option {
+	return newMtimeSortOption()
```

Comment:
Doesn't look like we really need `newMtimeSortOption()`
```suggestion
	return mtimeSortOption{}
```

Reply:
Yeah, we don't need it, removed newMtimeSortOption